internal/metrics/remote_write.go

```diff
@@ -113,38 +108,53 @@ func (w *writeClient) Write(ctx context.Context, data []byte) error {
 	}
 }
 
+func (r *RemoteWrite) IsEnabled() bool {
+	return r.client != nil
+}
+
+func (r *RemoteWrite) IsRunning() bool {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	return r.isRunning
+}
+
 func (r *RemoteWrite) Update(config models.DeviceConfigurationMessage) error {
-	// the following variables are only here until we implement reading from device configuration
-	serverURL := os.Getenv("REMOTEWRITE_SERVER_URL")
-	serverTimeout, _ := time.ParseDuration(os.Getenv("REMOTEWRITE_SERVER_TIMEOUT"))
-	if serverTimeout == 0 {
-		serverTimeout = defaultServerTimeout
-	}
-	requestNumSamples, _ := strconv.Atoi(os.Getenv("REMOTEWRITE_REQUEST_NUM_SAMPLES"))
-	if requestNumSamples == 0 {
-		requestNumSamples = r.RequestNumSamples // keep it unchanged
-	}
+	r.lock.Lock()
+	defer r.lock.Unlock()
 
-	// create configuration
-	r.config = RemoteWriteConfiguration{
-		Url:               serverURL,
-		Timeout:           serverTimeout,
-		RequestNumSamples: requestNumSamples,
+	// get new config
+	newConfig := (*models.MetricsReceiverConfiguration)(nil)
+	if config.Configuration != nil && config.Configuration.Metrics != nil {
+		newConfig = config.Configuration.Metrics.Receiver
 	}
 
-	if r.config.Url == "" { // feature disabled in development environment
-		return nil
+	// compare with old config
+	if r.Config == newConfig || (r.Config != nil && newConfig != nil && reflect.DeepEqual(r.Config, newConfig)) {
+		return nil // configuration did not change
 	}
 
-	// create client
-	if r.Client == nil {
-		serverURL, err := url.Parse(r.config.Url)
+	// use new config
+	featureDisabled := newConfig == nil || newConfig.URL == ""
+
+	if featureDisabled {
+		r.client = nil
+	} else {
+		serverURL, err := url.Parse(newConfig.URL)
 		if err != nil {
-			log.Error("failed creating metrics remote write client", err)
+			log.Error("metrics remote write configuration is invalid. Can not parse URL", err)
```

Comment:
is there a need to include URL in the error message so the errored value will be known?

Reply:
I rely on the fact that config manager already prints entire message

Reply:
added just in case...